### PR TITLE
 [FEATURE] #4 Fetch text/image for Card component

### PR DESCRIPTION
**Pull request by minh197**
## How to test: Locally
1. Create a new `.env` file in the `client` folder
2. `REACT_APP_API_KEY=7fbeff6f514f42f59fb852e44770e829`
You can get a new API key from the `https://newsapi.org/` as well (or you can use my API key to test this ticket)
3. Pull the change from my ticket
4. Run `npm start`
5. See that we are able to fetch the data from News API rendered it correctly to our card components
Screenshot 2024-03-24 at 2 27 44 PM

### CODE REVIEW

**Comment by minh197** [comment]
Pr is ready to review @Moethantko @naod1993

File: client/src/App.js [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
     <div className="App">
       <NavBar />
       <Routes>
         <Route path="/signup" element={<SignUp />} />
         <Route path="/login" element={<LogIn />} />
       </Routes>
-      <div>
-        <Card/>
-      </div>
+      <Card cardInfo={cardInfo}/>
```
Comment (Moethantko) [inline client/src/App.js]:
```suggestion
      <Route path="/" element={<Cards />} />
```

I think we don't really want logic in `App.tsx`. And, we want to display all the Cards in the home route ("/"). Cards component should have a child component `Card`.

File: client/src/components/Card/Card.jsx [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
-        </div>
-        <div className='card-content'>
-          <h2 className="m-0 p-0">Sample Card Name</h2>
-          <p className="date">March 16, 2024</p>
-          <p>This is card description</p>
-          <a className="read-more">Read more</a>
-        </div>
-      </div>
+    <div className="cards-container">
+      {cardInfo.map((info, index) => {
```
Comment (Moethantko) [inline client/src/components/Card/Card.jsx]: The logic for fetching `cardInfo` should be in this component, and this component should be renamed `Cards.jsx` in my opinion.

File: client/src/components/Card/Card.jsx [hunk 3]
```diff
@@ -1,22 +1,28 @@
...
+        return (
+          <div key={index} className="card">
```
Comment (Moethantko) [inline client/src/components/Card/Card.jsx]: I think this entire `div` should be in a seperate component, `Card.jsx`.

**Review by Moethantko — CHANGES_REQUESTED** [review]

File: client/src/App.js [hunk 4]
```diff
@@ -1,22 +1,36 @@
...
 import { Route, Routes } from "react-router-dom";
 import Card from "./components/Card/Card";
+import axios from 'axios';
+
 
 function App() {
+  const [cardInfo, setCardInfo] = useState([]);
+  const apiKey = process.env.REACT_APP_API_KEY;
+  useEffect(() => {
+    const fetchCardInfo = async() => {
```
Comment (Moethantko) [inline client/src/App.js]:
```suggestion
    (async() => {
      try {
        const response = await axios.get(`https://newsapi.org/v2/top-headlines?country=us&category=science&apiKey=${apiKey}`)
        setCardInfo(response.data.articles);
      } catch (error) {
        console.error("Error fetching news", error);
      }
    })()
```

Just a thought.

File: client/src/App.js [hunk 5]
```diff
@@ -4,33 +4,32 @@ import SignUp from "./components/Auth/SignUp";
...
 import { Route, Routes } from "react-router-dom";
-import Card from "./components/Card/Card";
+import Cards from "./components/Card/Cards";
 import axios from 'axios';
 
 
 function App() {
   const [cardInfo, setCardInfo] = useState([]);
   const apiKey = process.env.REACT_APP_API_KEY;
   useEffect(() => {
```
Comment (Moethantko) [inline client/src/App.js]: I also meant that this entire useEffect can be in `Cards.jsx`. Why do we need to pass `cardInfo` to `Cards.jsx` if we could fetch it in the component itself? I'm not sure if you have another reason for having API fetching code in `App.tsx.` But, I'd do it in the component.

**Review by Moethantko — CHANGES_REQUESTED** [review]
Just a question about the `useEffect` in `App.tsx`. Other than that, everything looks good!

File: client/src/App.js (hunk 5 above)
Comment (minh197) [reply client/src/App.js]: Adjusted accordingly. Pls review again!

**Review by Moethantko — APPROVED** [review]
Looks good!

**Pull request merged**